### ISSUE TITLE
Remove ExampleDS

### ISSUE THREAD

**Issue by LightAxe** [biemond/biemond-wildfly#157]
Is there a clean way to remove ExampleDS from the default JBoss install?  I don't see ensure=> absent as being viable, but I'll bet there's another way...

**Comment by jairojunior** [comment]
Other than convince RedHat not to include an example datasource in all configuration files? :smile: 

If you're hosting your own JBoss/Wildfly packages you could remove this configuration from your XMLs using augeas, XSLT or something like that.

If you're using roles/profiles, include this resource with ensure absent in your profile. I'd probably use this option, what's wrong with ensure => absent?

**Comment by LightAxe** [comment]
I tried:
```
  wildfly::datasources::datasource { 'ExampleDS':
    ensure => absent,
  }
```

....but it resulted in:

`Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Invalid parameter: 'ensure' on Wildfly::Datasources::Datasource[ExampleDS] at /etc/puppetlabs/code/environments/links/modules/profile/manifests/gwjboss.pp:89 on node gwappdevbc01`

**Comment by jairojunior** [comment]
You can do:

```puppet

  wildfly::util::resource { '/subsystem=datasources/data-source=ExampleDS':
    ensure => absent,
  }
```

**Comment by LightAxe** [comment]
I used:

```
  wildfly::util::resource { '/subsystem=datasources/data-source=ExampleDS':
    ensure => absent,
  }
```

and got:

```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Error while evaluating a Function Call, delete_undef_values(): expected an array or hash, got  type  String  at /etc/puppetlabs/code/environments/links/modules/wildfly/manifests/util/resource.pp:21:18 on node gwappdevbc01
```

**Comment by jairojunior** [comment]
Oh, you need na empty content:

content => {}

It should be the default value in wildfly::util::resource. Fixed in v1.0.0, but forgot to backport.

**Comment by LightAxe** [comment]
Ah, awesome.  Thanks so much!
